0038-http-auth-profiles/http-auth-profiles.md

```diff
@@ -0,0 +1,191 @@
+
+The profiles defined in this RFC should:
+
+* ...be as simple as possible while also being highly secure. 
+* ...clearly define any trade-offs involved in choosing a particular profile.
+* ...be easily implementable across programming languages in order to aid broad adoption.
+* ...utilise well-known and broadly deployed protocols in order to increase confidence in the profiles.
+* ...be suitable for high-performance use-cases, such as ILPv4 packet processing.
+* ...resist replays (emphasis: resist, not necessarily be replay-proof, in order to support high-performance use-cases).
+* ...not negate or prevent the ability of operators to layer-on more security "on top of" a profile defined in this RFC.
+
+## Specification
+
+### Overview
+This document defines three authentication profiles for securing HTTP network connections in Interledger.
+
+### Authentication
+When authenticating requests between Interledger nodes, it is important to choose an algorithm that maintains an appropriate balance between usability and security, while at the same time maintaining high-performance _and_ interoperability.
+
+In order to find this balance, this document defines three Authentication profiles, each with various trade-offs that should be considered before use:
+
+* `SIMPLE`: Allows two ILP nodes to utilize a previously agreed-upon shared-secret as a [Bearer token](https://tools.ietf.org/html/rfc6750) in all HTTP requests. Peers SHOULD consider this token to be opaque and SHOULD NOT derive any special meaning from the token. 
+
+* `JWT_HS_256`: Allows two ILP nodes to utilize a previously agreed-upon shared-secret in order to _derive_ a `Bearer token` that conforms to the JSON Web Token (JWT) specification as defined in [RFC-7519](https://www.rfc-editor.org/rfc/rfc7519.html) using the `HS_256` signing algorithm defined in section 3.2 of [RFC-7518](https://www.rfc-editor.org/rfc/rfc7518#section-3.2).
+
+* `JWT_RS_256`: Allows two ILP nodes to utilize public-key pairs to _derive_ a `Bearer token` that conforms to the JSON Web Token (JWT) specification as defined in [RFC-7519](https://www.rfc-editor.org/rfc/rfc7519.html) using the `RS_256` signing algorithm defined in section 3.3 of [RFC-7518](https://www.rfc-editor.org/rfc/rfc7518#section-3.3).
+
+Peers MAY use any standard HTTP authentication mechanism to authenticate incoming requests, but SHOULD support `JWT_HS_256` at a minimum. It is RECOMMENDED to use `JWT_RS_256` for production deployments.
+
+#### `SIMPLE` Authentication Profile
+This profile allows two ILP nodes to utilize a previously agreed-upon shared-secret that contains at least 32 bytes (256 bits) of randomly generated data, and is encoded using Base64.
+
+Because tokens in this profile do not inherently contain information about the identity of the caller, requests MUST contain an additional HTTP request-header named `Auth-Principal`.
```

Comment:
What's the advantage of using this header over including the account or user identifier in the token, or using HTTP Basic Auth?

Reply:
> What's the advantage of using this header over including the account or user identifier in the token,

The advantages are minor IMO, but 1) it's explicit, so two peers don't have to invent a new token-scheme. This helps interop. 2) It allows the actual auth token to be opaque. If two peers need to infer meaning into the token, it's not really opaque anymore.

>  or using HTTP Basic Auth?

The disadvantage of `BASIC Auth` is that it's not a Bearer token scheme. This may not matter depending on what we decide here, but at one point in time I was able to peer a Java implementation that only implements `JWT_HS_256` with the prior versions of the Rust Connector without issue because I could generate a `JWT_HS_256` token, and just configure the Rust connector to use that as an opaque token.

Reply:
This header is now removed due to the new `SIMPLE` verbiage.